### PR TITLE
Adding documentation for style guide, contribution and getting started

### PR DESCRIPTION
**Pull request by dpetrisko**

### CODE REVIEW

**Comment by dpetrisko** [comment]
I would recommend looking at the actual repo instead of the files_changed, since Markdown diffs are hard to parse (for me, at least).

File: GETTING_STARTED.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
+# Getting started
+    git clone https://github.com/black-parrot/pre-alpha-release.git
+    # make tools will clone, build and install toolchains needed for BlackParrot
+    # For faster builds, make tools -j is parallelizable!
+    make tools
+    # make tidy_tools will remove all tool build directories, while leaving the installation
+    # make progs will build all open source test programs that ship with BlackParrot
+    make progs
+    # make ucode will build the CCE microcode used for BlackParrot's coherence engine
```
Comment (muwyse) [inline GETTING_STARTED.md]: Perhaps add a note about rebuilding the microcode when the CCE ucode files get updated. We could also omit this since we will eventually add the "ucode on demand" feature to building.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+### Bug reports
+**Something doesn't look right to me.**
+- Suggested fields:
+    - Descriptive title.
+    - Module/tool/makefile/script in question.
+    - Current behavior (steps for reproduction).
+    - Expected behavior.
+    - Proposed solution(s), including scope of work and affected modules.
+    - Methods of verification.
+- Example: Power waste by propagating data when stalled.
```
Comment (muwyse) [inline CONTRIBUTING.md]: Example doesn't seem to be "complete", in that we describe a bunch of things the issue should specify, but our example is a single sentence.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+### Feature requests
+**I would like BlackParrot to do this.**
+- Suggested fields:
+    - Descriptive title.
+    - Module/tool/makefile/script in question.
+    - Desired behavior / rationale for change.
+    - Proposed solution(s), including scope of work and affected modules.
+    - Methods of quantitative evaluation.
+- Example: Support pre-fetching in I$.
```
Comment (muwyse) [inline CONTRIBUTING.md]: same as above.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+### Informational
+**I don't understand this.**
+- Suggested fields:
+    - Descriptive title.
+    - Module/tool/makefile/script in question.
+    - Information requested.
+    - Background information already gathered.
+- Example: What algorithm does D$ LRU use?
```
Comment (muwyse) [inline CONTRIBUTING.md]: same as above.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+The absolute most important factor in making a successful open-source contribution is communication with the maintainers. Therefore it is highly recommended to raise an issue and discuss potential implementation paths before making a pull request. "Dumping code" is discouraged. 
+
+An ideal collaboration looks like the following:
+1. I would like BlackParrot to be able to eat ants.
+2. Open an issue
+    - Title: "Support for eating ants"
+    - Supporting eating ants would increase PPA because BlackParrot would have more energy.
+    - We could either modify the beak or the talons, with the tradeoff being that the beak is more efficent, but harder to modify.
+3. Communicate with BlackParrot maintainers
+    - We have an extensive talon modification pending on another branch, which will be incompatible with your proposal. Your beak proposal interesting though, have you considered the effects on eating berries?
```
Comment (muwyse) [inline CONTRIBUTING.md]: "...beak proposal *is* interesting..."

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+    - Functionality, power, performance CI MUST pass before merge is accepted
+- dev: development branch  
+    - Developers should be able to git clone and run a buggy copy at all times.
+    - CI should pass before merge is accepted.  
+    - Occasional churn at this level is acceptable as there may be inter-module dependencies being worked on.
+- top_dev: bp_top development branch
+    - Top level wrapping modules; tiles, off-chip networks, SoC configurations.
+    - CI should often pass, but there is no gatekeeper at this level.  
+- fe_dev, be_dev, me_dev: Module level development branch
+    - All \<end\> work should branch off of and merge into this branch. fe_dev will periodically merge from and into dev.
```
Comment (muwyse) [inline CONTRIBUTING.md]: "fe_dev, be_dev, and me_dev will periodically merge from and into dev."

Not just fe_dev, right?

File: STYLE_GUIDE.md [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+            x = 3;  // and optional alignment of x = statement.
+        else
+          begin
+            x = 1;
+	    	y = 0;
+          end
+
+- Use localparam keyword for derived parameters and input parameters that should not be set by the party instantiating the module.
+- Localparams in port lists are allowed, but only for sizing ports. Other generated localparams should go near their consumers.
+- Prefer input [1:0] to input logic [1:0] and output [1:0] to output logic [1:0]
```
Comment (muwyse) [inline STYLE_GUIDE.md]: I'm not sure we follow this. Most files use "output logic", I think.

File: STYLE_GUIDE.md [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+    
+        foo
+         #(.bar_p(foo_bar_p))
+         inst
+          (.baz_i(foo_baz_li));
+- Signals corresponding to a pipeline stage in a module which has multiple stages should have a pipeline stage suffix
+    - foobar_ex1_lo
+    - barbaz_tv_n
+- ‘0’, ‘1’, ‘2’, and ‘8’ (when used as byte width) are the only allowable magic numbers. Else, consider strongly the use of a localparam.
+- Prefer packages over \`includes. While \`defines are global and so this makes no difference to the LRM, BlackParrot style is to compile all packages first, so most tools will include definitions in the correct order.
```
Comment (muwyse) [inline STYLE_GUIDE.md]: Grammar is a little funky here, revise.

Comment (dpetrisko) [reply STYLE_GUIDE.md]: Revised, let me know if this makes more sense

File: STYLE_GUIDE.md (hunk 7 above)
Comment (dpetrisko) [reply STYLE_GUIDE.md]: Yeahh, it's definitely not a well-followed rule (Prefer vs Use).  I think it's still valuable for new code, but I added in some more justification.

File: CONTRIBUTING.md (hunk 6 above)
Comment (dpetrisko) [reply CONTRIBUTING.md]: fixed

File: CONTRIBUTING.md (hunk 5 above)
Comment (dpetrisko) [reply CONTRIBUTING.md]: fixed

File: GETTING_STARTED.md (hunk 1 above)
Comment (dpetrisko) [reply GETTING_STARTED.md]: Added

File: CONTRIBUTING.md (hunk 2 above)
Comment (dpetrisko) [reply CONTRIBUTING.md]: Added complete examples
